agent/utils/filesystem.go

```diff
@@ -1,17 +1,36 @@
 package utils
 
 import (
+	"errors"
 	"fmt"
 	"io/ioutil"
+	"os"
+	"strings"
 )
 
 const (
-	mountIdLocation          = "/var/lib/docker/image/overlay2/layerdb/mounts/%s/mount-id"
-	targetFileSystemLocation = "/var/lib/docker/overlay2/%s/merged"
+	overlayFsMountIdLocation = "/var/lib/docker/image/overlay2/layerdb/mounts/%s/mount-id"
+	// The actual path on the host is /run/containerd/io... but we don't change the path within the agent to not break backwards compatibility with the flag
+	containerdMountIdLocation = "/var/lib/docker/io.containerd.runtime.v2.task/k8s.io/%s/rootfs"
+	targetFileSystemLocation  = "/var/lib/docker/overlay2/%s/merged"
 )
 
 func GetTargetFileSystemLocation(containerId string) (string, error) {
-	fileName := fmt.Sprintf(mountIdLocation, containerId)
+	fileName := fmt.Sprintf(overlayFsMountIdLocation, containerId)
+
+	// Check if the node uses overlayfs mount location
+	if _, err := os.Stat(fileName); errors.Is(err, os.ErrNotExist) {
```

Comment:
I don't know which error I get here, but checking for `os.ErrNotExist` is a bit too specific for my environment (Azure Kubernetes Services). However, if I check for `!= nil`, it works.

```suggestion
	if _, err := os.Stat(fileName); err != nil {
```